### PR TITLE
MINOR: Upgrade Gradle wrapper to version 8.4

### PR DESCRIPTION
**Pull request by bmscomp**
This pull request is aiming to upgrade version to Gradle to last stable one 8.4 

This version brings up support for JDK 21 
More then that it fixes two security issues with much bug fixing related to the previous releases 

[Incorrect permission assignment for symlinked files used in copy or archiving operations](https://github.com/gradle/gradle/security/advisories/GHSA-43r3-pqhv-f7h9)
[Possible local text file exfiltration by XML External entity injection](https://github.com/gradle/gradle/security/advisories/GHSA-mrff-q8qj-xvg8)

For more information please refer to https://docs.gradle.org/8.4/release-notes.html

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -214,11 +215,11 @@ fi
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
 DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
 
-# Collect all arguments for the java command;
-#   * $DEFAULT_JVM_OPTS, $JAVA_OPTS, and $GRADLE_OPTS can contain fragments of
-#     shell script including quotes and variable substitutions, so put them in
-#     double quotes to make sure that they get re-expanded; and
-#   * put everything else in single quotes, so that it's not re-expanded.
+# Collect all arguments for the java command:
+#   * DEFAULT_JVM_OPTS, JAVA_OPTS, JAVA_OPTS, and optsEnvironmentVar are not allowed to contain shell fragments,
```
Comment (divijvaidya) [inline gradlew]: JAVA_OPTS is repeated. 

Why did we remove GRADLE_OPTS? It a legitimate environment variable: https://docs.gradle.org/current/userguide/build_environment.html

File: gradlew [hunk 2]
```diff
@@ -157,15 +158,15 @@ if ! "$cygwin" && ! "$darwin" && ! "$nonstop" ; then
...
+        # shellcheck disable=SC2039,SC3045
         MAX_FD=$( ulimit -H -n ) ||
             warn "Could not query maximum file descriptor limit"
     esac
     case $MAX_FD in  #(
       '' | soft) :;; #(
       *)
         # In POSIX sh, ulimit -n is undefined. That's why the result is checked to see if it worked.
-        # shellcheck disable=SC3045
+        # shellcheck disable=SC2039,SC3045
```
Comment (divijvaidya) [inline gradlew]: Seems like SC2039 is retired as per https://www.shellcheck.net/wiki/SC2039. Why did we add this?

File: gradlew (hunk 1 above)
Comment (bmscomp) [reply gradlew]: @divijvaidya  thanks so much for this catch, I used to think that is is a missing from the Gradle wrapper update script I already restored this value in the last commit

File: gradlew (hunk 2 above)
Comment (bmscomp) [reply gradlew]: @divijvaidya  I restored the default original value

File: gradlew [hunk 3]
```diff
@@ -214,11 +215,11 @@ fi
...
-# Collect all arguments for the java command;
-#   * $DEFAULT_JVM_OPTS, $JAVA_OPTS, and $GRADLE_OPTS can contain fragments of
-#     shell script including quotes and variable substitutions, so put them in
-#     double quotes to make sure that they get re-expanded; and
-#   * put everything else in single quotes, so that it's not re-expanded.
+# Collect all arguments for the java command:
+#   * DEFAULT_JVM_OPTS, JAVA_OPTS, GRADLE_OPTS, and optsEnvironmentVar are not allowed to contain shell fragments,
+#     and any embedded shellness will be escaped.
+#   * For example: A user cannot expect ${Hostname} to be expanded, as it is an environment variable and will be
+#     treated as '${Hostname}' itself on the command line.
```
Comment (ijuma) [inline gradlew:222]: Why does this look different from https://github.com/apache/kafka/pull/14530/files? How did you update this file?

Comment (bmscomp) [reply gradlew:222]: @ijuma  The file is generated by gradle itself , I just fixed I did not myself remove the GRADLE_OPTS

Comment (ijuma) [reply gradlew:222]: The version I linked was also generated by gradle. What command did you run?

**Comment by chia7712** [comment]
close this PR as version of gradle is 8.6 now

**Pull request closed without merging**